cime_config/buildnml

```diff
@@ -233,6 +235,14 @@ def buildnml(case, caseroot, compname):
                     "WARNING: Do you have appropriate initial conditions for this simulation?"
                     + " Check that the finidat file used in the lnd_in namelist is appropriately spunup for your case"
                 )
+        if "PLUMBER2" in clm_usrdat_name and clm_force_coldstart == "off":
```

Comment:
I'm thinking rather than copy and paste this whole section, that the if just be changed to do the same thing for PLUMBER2 and NEON.

Reply:
These do have different name formats that are used, eg `\w\w-\w\w\w\` for PLUMBER2 and `\w\w\w\w\` for NEON, so I'd slightly prefer to keep these separate.

If, with that additional bit of information, you still think this is necessary, I could do something like this (though I think it would be more confusing than helpful):
```
if ("NEON" in clm_usrdat_name) or ("PLUMBER2" in clm_usrdat_name) and clm_force_coldstart == "off":
            if "NEON" in clm_usrdat_name:
                 match = '\w\w\w\w\'
            if "PLUMBER2" in clm_usrdat_name:
                 match = '\w\w-\w\w\w\
            if ("_transient" in clm_nml_use_case) and (
                re.fullmatch(r"{}.transient".format(match), casename) is None
                or clm_usrdat_name is "NEON.PRISM"
            ):
                logger.warning(
                    "WARNING: Do you have appropriate initial conditions for this simulation?"
                    + " Check that the finidat file used in the lnd_in namelist is appropriately spunup for your case"
                )
```

Reply:
Ahh, I see a different match.

Because of the reduced duplication I personally prefer the code you propose. Reducing duplication is really important for code maintenance.  So I like your proposal. Are you warm enough to the proposal that you could bring it in? Maybe adding a comment to explain what's going on would help if you find it confusing?

Code readability and trying to not be confusing are important. I think the best way to do that for buildnml is to break it up into smaller subroutines and have python unit testing for them. That's obviously a longer term thing though.

Reply:
Ok, I removed some of the duplication. I ended up leaving the line with match as one line rather than using the 'match' variable because the backslashes were causing some issues when it was interpreted. But, I think this is an improvement.